src/Queue/Jobs/RabbitMQJob.php

```diff
@@ -75,15 +75,14 @@ public function attempts(): int
         $headers = Arr::get($this->message->get_properties(), 'application_headers');
 
         if (! $headers) {
-            return 0;
+            return 1;
         }
 
         $data = $headers->getNativeData();
 
         $laravelAttempts = (int) Arr::get($data, 'laravel.attempts', 0);
-        $xDeathCount = (int) Arr::get($headers->getNativeData(), 'x-death.0.count', 0);
 
-        return $laravelAttempts + $xDeathCount;
+        return $laravelAttempts + 1;
```

Comment:
x-death count is used when job has been rejected, nacked or requeued. It should be taken into consideration

Reply:
My solution does not care about nacked messages. Ofcourse you have to tell rabbitMQ what to do with the message, when the job is released. But because the DLX was removed on the main Queue, rabbitMQ does nothing with the message and the message is gone after nacking or acknowledging.

My solution always pushes a 'new' message with different attributes/properties to the queue.

See also my comment in issue: #305

Reply:
Maybe there is an solution for not loosing the messages, when no DLX is set and rejecting messages and keeping the xDeathCount property into account.

I have to test this situation.

CASE:
define queue without DLX.

When rejecting also requeue the message, by calling `$this->reject($this, true)`.

Hopefully xDeathCount is updated when nacking the message. 

The only problem i think persistst: is that when there are 1000 Jobs in the queue, the failed Job will proccess only when the other 999 are processed first. Leaving us with a situation that we can't tell when in time the message is processed again.

Reply:
result:  
xDeathCount is only set when DLX is used. 
DLX set to its own queue results in errors. DLX was not designed to route messages to it self (as far as i know).

When removing DLX on the main queue (in your code) we cant use `reject()`, or message will be lost.